src/bun.js/bindings/BunProcess.cpp

```diff
@@ -1127,40 +1205,89 @@ JSC_DEFINE_HOST_FUNCTION(Process_emitWarning, (JSGlobalObject * lexicalGlobalObj
     Zig::GlobalObject* globalObject = jsCast<Zig::GlobalObject*>(lexicalGlobalObject);
     VM& vm = globalObject->vm();
     auto scope = DECLARE_THROW_SCOPE(vm);
+    auto* process = jsCast<Process*>(globalObject->processObject());
 
-    if (callFrame->argumentCount() < 1) {
-        throwVMError(globalObject, scope, "Not enough arguments"_s);
-        return {};
+    auto warning = callFrame->argument(0);
+    auto type = callFrame->argument(1);
+    auto code = callFrame->argument(2);
+    auto ctor = callFrame->argument(3);
+    auto detail = jsUndefined();
+
+    auto dep_warning = jsString(vm, String("DeprecationWarning"_s));
+
+    if (Bun__Node__ProcessNoDeprecation && JSC::JSValue::strictEqual(globalObject, type, dep_warning)) {
+        return JSValue::encode(jsUndefined());
     }
 
-    RETURN_IF_EXCEPTION(scope, {});
+    if (!type.isNull() && type.isObject() && !isJSArray(type)) {
+        ctor = type.get(globalObject, Identifier::fromString(vm, "ctor"_s));
+        RETURN_IF_EXCEPTION(scope, {});
 
-    auto* process = jsCast<Process*>(globalObject->processObject());
+        code = type.get(globalObject, builtinNames(vm).codePublicName());
+        RETURN_IF_EXCEPTION(scope, {});
 
-    JSObject* errorInstance = ([&]() -> JSObject* {
-        JSValue arg0 = callFrame->uncheckedArgument(0);
-        if (!arg0.isEmpty() && arg0.isCell() && arg0.asCell()->type() == ErrorInstanceType) {
-            return arg0.getObject();
-        }
+        detail = type.get(globalObject, vm.propertyNames->detail);
+        RETURN_IF_EXCEPTION(scope, {});
+        if (!detail.isString()) detail = jsUndefined();
 
-        WTF::String str = arg0.toWTFString(globalObject);
-        auto err = createError(globalObject, str);
-        err->putDirect(vm, vm.propertyNames->name, jsString(vm, String("warn"_s)), JSC::PropertyAttribute::DontEnum | 0);
-        return err;
-    })();
+        type = type.get(globalObject, vm.propertyNames->type);
+        RETURN_IF_EXCEPTION(scope, {});
+        if (!type.toBoolean(globalObject)) type = jsString(vm, String("Warning"_s));
+    } else if (type.isCallable()) {
+        ctor = type;
+        code = jsUndefined();
+        type = jsString(vm, String("Warning"_s));
```

Comment:
ditto

Reply:
where was this resolved?

Reply:
this value is passed to `putDirect` unmodified so it needs to be a JSValue